### ISSUE TITLE
Beaker provisioning fails if recipesets are not named on 1.5.3 

### ISSUE THREAD

**Issue by jaypoulz** [CentOS-PaaS-SIG/linchpin#575]
As discovered this morning:

If you attempt beaker provisioning on 1.5.3 without a name for you recipe set, you end up encountering:

```
fatal: [localhost]: FAILED! => {"msg": "the field 'args' has an invalid value ([]), and could not be converted to an dict.The error was: dictionary update sequence element #0 has length 1; 2 is required\n\nThe error appears to have been in '/home/jenkins/envs/provisioner/lib/python2.7/site-packages/linchpin/provision/roles/beaker/tasks/provision_resource_group.yml': line 79, column 5, but may\nbe elsewhere in the file depending on the exact syntax problem.\n\nThe offending line appears to be:\n\n\n  - name: \"combine bkr_hosts and bkr_names\"\n    ^ here\n"}
```

This is most likely because some funny business happens with bkr_names:
```
ok: [localhost] => (item={'count': 1, 'arch': u'x86_64', 'variant': u'Server', 'distro': u'RHEL-7.5'}) => {"ansible_facts": {"bkr_names": "[] + [{'name': Undefined}]"}, "brs": {"arch": "x86_64", "count": 1, "distro": "RHEL-7.5", "variant": "Server"}, "changed": false}
```

The irony of the issue is that with pre-1.5.3 versions of linchpin, adding "name" as part of recipesets means the template won't validate.

**Comment by p3ck** [comment]
I am not able to reproduce this with 1.5.3  If I go back to 1.5.2 it fails schema.
@jaypoulz Can you try the latest 1.5.4 and see if you still have the same issue?

**Comment by jaypoulz** [comment]
It brings me no joy in being the bearer of bad news, but yes, this problem is still present in 1.5.4.

Relevant log lines appear to be:
```
18:24:44 [x86_64] TASK [beaker : set bkr_names] **************************************************
18:24:44 [x86_64] ok: [localhost] => (item={'count': 1, 'variant': u'Server', 'arch': u'x86_64', 'distro': u'RHEL-7.4'}) => {"ansible_facts": {"bkr_names": "[] + [{'name': Undefined}]"}, "brs": {"arch": "x86_64", "count": 1, "distro": "RHEL-7.4", "variant": "Server"}, "changed": false}
```
and
```
18:24:44 [x86_64] TASK [beaker : combine bkr_hosts and bkr_names] ********************************
18:24:44 [x86_64] fatal: [localhost]: FAILED! => {"msg": "the field 'args' has an invalid value ([]), and could not be converted to an dict.The error was: dictionary update sequence element #0 has length 1; 2 is required\n\nThe error appears to have been in '/home/jenkins/envs/provisioner/lib/python2.7/site-packages/linchpin/provision/roles/beaker/tasks/provision_resource_group.yml': line 79, column 5, but may\nbe elsewhere in the file depending on the exact syntax problem.\n\nThe offending line appears to be:\n\n\n  - name: \"combine bkr_hosts and bkr_names\"\n    ^ here\n"}
```

My PinFile looks like:
---
jenkins-slave:
  topology:
    topology_name: jenkins-slave
    resource_groups:
    - resource_group_name: beaker-slaves
      resource_group_type: beaker
      resource_definitions:
      - role: bkr_server
        job_group: {{ job_group }}
        whiteboard: "{{ arch }} slave for multiarch testing"
        # options to configure amount of time spent polling beaker:
        # 180 attempts with 60 seconds wait time between them,
        # provisioning timeout is roughly 10800 seconds
        max_attempts: 180
        attempt_wait_time: 60
        # option to set cancellation message if linchpin cancels provisioning
        cancel_message: Job cancelled due to beaker request timeout
        recipesets:
        - distro: "RHEL-7.4"
          arch: {{ arch }}
          variant: Server
          count: 1
  layout:
    inventory_layout:
      hosts:
        jenkins-slave:
          count: 1
          host_groups:
            - rhel7
            - certificate_authority
            - repositories
            - jenkins_slave
            - master_node
```

To make it work, just add `name: "Jenkins Slave" in the recipesets object. E.g.
```
        recipesets:
        - distro: "RHEL-7.4"
          arch: {{ arch }}
          variant: Server
          count: 1
          name: "Jenkins Slave"
```

**Comment by herlo** [comment]
@samvarankashyap I think something like this might address the issue.

```
--- a/linchpin/provision/roles/beaker/tasks/provision_resource_group.yml
+++ b/linchpin/provision/roles/beaker/tasks/provision_resource_group.yml
@@ -80,6 +80,12 @@
   - name: "combine bkr_hosts and bkr_names"
     set_fact:
       bkr_host_values: "{{ bkr_hosts | combine_hosts_names(bkr_names) }}"
+    when: bkr_names is defined
+
+  - name: "combine bkr_hosts and bkr_names"
+    set_fact:
+      bkr_host_values: "{{ bkr_hosts }}"
+    when: bkr_names is not defined

```

**Comment by herlo** [comment]
Resolved by #629
